src/libraries/System.Transactions.Local/src/System/Transactions/Oletx/OletxTransaction.cs

```diff
@@ -1133,12 +1127,13 @@ internal void FireOutcome(TransactionStatus statusArg)
                 }
             }
 
+#pragma warning disable IDE0031 // Null check can be simplified
             // Let the InternalTransaciton know about the outcome.
             if (InternalTransaction != null)
             {
                 InternalTransaction.DistributedTransactionOutcome(InternalTransaction, Status);
             }
-
+#pragma warning restore IDE0031
```

Comment:
Why does this one need to be suppressed rather than:
```C#
InternalTransaction?.DistributedTransactionOutcome(InternalTransaction, Status);
```
?

Reply:
The method `DistributedTransactionOutcome` is a static method. `InternalTransaction` is the name of both a property and the type of that property. Fixing this to use `?.` changes the invocation to try to invoke an instance method on the `InternalTransaction` property value instead of the `InternalTransaction` type.

This one definitely deserves a Roslyn issue.

Reply:
Ah, then yeah, that's a bug.